client/extensions/woocommerce/app/settings/payments/payment-method-item.js

```diff
@@ -1,76 +1,163 @@
 import ListItemField from 'woocommerce/components/list/list-item-field';
+import PaymentMethodEdit from './payment-method-edit';
+import { savePaymentMethod } from 'woocommerce/state/sites/payment-methods/actions';
 
-const PaymentMethodItem = ( { method, translate, onCancel, onEdit, currentlyEditingId } ) => {
-	let editButtonText = translate( 'Set up' );
-	if ( currentlyEditingId === method.id ) {
-		editButtonText = translate( 'Cancel' );
-	}
+class PaymentMethodItem extends Component {
+	static propTypes = {
+		closeEditingPaymentMethod: PropTypes.func.isRequired,
+		currentlyEditingId: PropTypes.string,
+		currentlyEditingMethod: PropTypes.shape( {
+			id: PropTypes.string,
+		} ),
+		method: PropTypes.shape( {
+			title: PropTypes.string.isRequired,
+			isSuggested: PropTypes.bool,
+			fees: PropTypes.string,
+			id: PropTypes.string,
+			informationUrl: PropTypes.string,
+		} ),
+		openPaymentMethodForEdit: PropTypes.func.isRequired,
+		savePaymentMethod: PropTypes.func.isRequired,
+	};
 
-	const onEditHandler = () => {
+	onEditHandler = () => {
+		const { method } = this.props;
+		const currentlyEditingId = this.props.currentlyEditingMethod &&
+			this.props.currentlyEditingMethod.id;
 		if ( currentlyEditingId === method.id ) {
-			onCancel( method );
+			this.onCancel( method );
 		} else {
-			onEdit( method );
+			this.onEdit( method );
 		}
 	};
 
-	return (
-		<ListItem>
-			<ListItemField>
-				{
-					method.isSuggested &&
-					(
-						<p className="payments__method-suggested">
-							{ translate( 'Suggested Method' ) }
+	onCancel = () => {
+		const { method } = this.props;
+		this.props.closeEditingPaymentMethod( this.props.site.ID, method.id );
+	}
+
+	onEdit = ( method ) => {
+		const { site } = this.props;
+		this.props.openPaymentMethodForEdit( site.ID, method.id );
+	}
+
+	onEditField = ( field, value ) => {
+		this.props.changePaymentMethodField( this.props.site.ID, field, value );
+	}
+
+	onSave = ( method ) => {
+		const { site, translate } = this.props;
+
+		const successAction = () => {
+			this.props.closeEditingPaymentMethod( site.ID, method.id );
+			return successNotice(
+				translate( 'Payment method successfully saved.' ),
+				{ duration: 4000 }
+			);
+		};
+
+		const errorAction = () => {
+			return errorNotice(
+				translate( 'There was a problem saving the payment method. Please try again.' )
+			);
+		};
+
+		this.props.savePaymentMethod( site.ID, method, successAction, errorAction );
+	}
+
+	render() {
+		const currentlyEditingId = this.props.currentlyEditingMethod &&
+			this.props.currentlyEditingMethod.id;
+		const { method, translate } = this.props;
+		let editButtonText = translate( 'Set up' );
+		if ( currentlyEditingId === method.id ) {
+			editButtonText = translate( 'Cancel' );
+		}
+
+		return (
+			<ListItem>
+				<ListItemField>
+					{
+						method.isSuggested &&
+						(
+							<p className="payments__method-suggested">
+								{ translate( 'Suggested Method' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **73** times:
`translate( 'Suggested media' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).